drivers/mtd_sdmmc/mtd_sdmmc.c

```diff
@@ -141,35 +141,8 @@ static int mtd_sdmmc_write_page(mtd_dev_t *dev, const void *buff, uint32_t page,
 
 static int mtd_sdmmc_erase_sector(mtd_dev_t *dev, uint32_t sector, uint32_t count)
 {
-#if IS_ACTIVE(CONFIG_MTD_SDMMC_ERASE) && IS_USED(MODULE_MTD_WRITE_PAGE)
     mtd_sdmmc_t *mtd_sd = (mtd_sdmmc_t*)dev;
-
-    DEBUG("mtd_sdmmc_erase_sector: sector: %" PRIu32 " count: %" PRIu32 "\n",
-          sector, count);
-
-    if (dev->work_area == NULL) {
-        DEBUG("mtd_sdmmc_erase_sector: no work area\n");
-        return -ENOTSUP;
-    }
-    memset(dev->work_area, 0, SDMMC_SDHC_BLOCK_SIZE);
-    while (count) {
-        if (sdmmc_write_blocks(mtd_sd->sdmmc, sector, SDMMC_SDHC_BLOCK_SIZE,
-                               1, dev->work_area, NULL)) {
-            return -EIO;
-        }
-        --count;
-        ++sector;
-    }
```

Comment:
What I don't understand is why you added this fallback implementation when you have a proper `sdmmc_erase_blocks()`.

This was only needed for `sdcard_spi` because it does not implement the erase command at all.

Reply:
> What I don't understand is why you added this fallback implementation when you have a proper `sdmmc_erase_blocks()`.

To be honest, I don't remember. Probably, I just used `mtd_sdcard_spi` as a template to implement `mtd_sdmmc` so that it is just a copy of it. I didn't know the `MTD_DRIVER_FLAG_DIRECT_WRITE` flag.